### PR TITLE
Test tools shipped with Galaxy

### PR DESCRIPTION
**Pull request by mvdbeek**
This fixes tests for a lot of the tools we ship with galaxy. I've set up a Jenkins job for these as well.

xref. #1715

### CODE REVIEW

**Comment by mvdbeek** [comment]
There are some accompanying test data changes in https://github.com/galaxyproject/galaxy-test-data/pull/1

File: lib/galaxy/jobs/output_checker.py [hunk 1]
```diff
@@ -130,12 +133,15 @@ def check_output(tool, stdout, stderr, tool_exit_code, job):
...
-                peak = stderr[0:250]
-                log.debug("Tool produced standard error failing job - [%s]" % peak)
                 state = DETECTED_JOB_STATE.GENERIC_ERROR
             else:
                 state = DETECTED_JOB_STATE.OK
 
+        if DETECTED_JOB_STATE != DETECTED_JOB_STATE.OK and stderr:
+            if stderr:
+                peak = stderr[0:ERROR_PEAK]
+                log.debug("job failed, standard error is - [%s]" % peak)
```
Comment (jmchilton) [inline lib/galaxy/jobs/output_checker.py]: I change this locally a few times of week and have never bothered to push it upstream. This file is always an open tab - thanks for the change!

**Comment by mvdbeek** [comment]
```
 113 passed, 1 deselected, 1 warnings in 1502.15 seconds 
```

yay, just a single tool that needed a cheetah fix (tools/filters/wc_gnu.xml)!

File: lib/galaxy/jobs/output_checker.py [hunk 2]
```diff
@@ -15,6 +15,9 @@
 )
 
 
+ERROR_PEAK = 2000
```
Comment (martenson) [inline lib/galaxy/jobs/output_checker.py]: could the 8x larger peak be a problem for the log size?

Comment (mvdbeek) [reply lib/galaxy/jobs/output_checker.py]: We can make it configurable. But 2000 is still less than your average traceback.

**Comment by mvdbeek** [comment]
Cool, the tests are passing on Jenkins: https://jenkins.galaxyproject.org/view/All/job/docker-main-tools/lastCompletedBuild/testReport/test.functional/test_toolbox_pytest/

**Comment by mvdbeek** [comment]
Seems like I've missed one tool: https://jenkins.galaxyproject.org/view/All/job/docker-main-tools-py3/1/testReport/junit/test.functional/test_toolbox_pytest/

File: tools/stats/gsummary.xml [hunk 3]
```diff
@@ -4,7 +4,7 @@
     <edam_topic>topic_2269</edam_topic>
   </edam_topics>
   <requirements>
-    <requirement type="package" version="2.7.8">rpy2</requirement>
+    <requirement type="package" version="2.9.4">rpy2</requirement>
```
Comment (jmchilton) [inline tools/stats/gsummary.xml:7]: Tool version update?

Comment (mvdbeek) [reply tools/stats/gsummary.xml:7]: Sure, done that in https://github.com/galaxyproject/galaxy/pull/6627/commits/c939e7bb616103fb4d79067cf3d8159e276524ea

**Comment by mvdbeek** [comment]
Alright, the python3 tests are passing as well: https://jenkins.galaxyproject.org/view/All/job/docker-main-tools-py3/2/

**Comment by jmchilton** [comment]
Fantastic - thanks so much!

**Pull request merged**